File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
   
    ans =  []int{}
	dfs(root  ) 
	return ans 
}

var ans []int

func dfs(node *TreeNode ){

	if node != nil {
		
		dfs(node.Left  ) 
		ans = append(ans, node.Val) 
        dfs(node.Right )  		

	}

}

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        
        vector<int> ans; 
        dfs(root ,ans);
        return ans; 
        
    }
    void dfs(TreeNode* node , vector<int>& ans){
        if ( node != nullptr){
            
            dfs(node->left , ans ) ;
            ans.push_back(node->val) ; 
            dfs(node->right , ans ) ;
        }
    }
};